rwengine/include/script/ScriptTypes.hpp

```diff
@@ -66,13 +68,19 @@ struct SCMOpcodeParameter {
 
 	int integerValue() const 
 	{
-		if ( type == TGlobal )
-		{
-			return * globalInteger;
-		}
-		else
+		switch (type)
 		{
-			return integer;
+		case TGlobal:
+			return *globalInteger;
+		case TInt8:
+			return (uint8_t)integer;
+		case TInt16:
+			return (uint16_t)integer;
+		case TInt32:
+			return (uint32_t)integer;
```

Comment:
[Looks like I changed these to signed ints in my commit](https://github.com/JayFoxRox/openrw/pull/11/commits/5455d2912675419032292442732c6c759813b393)

I can't remember where it mattered specifically, but I probably did it for a reason.
I didn't see any forced signed comparisions in SCM yet either so it's probably signed by default.